ui_testing/testcases/basic_tests/test04_testunits.py

```diff
@@ -812,3 +812,51 @@ def test025_edit_approach_overview_button(self):
         self.assertEqual(self.base_selenium.get_url(), '{}testUnits'.format(self.base_selenium.url))
         self.base_selenium.LOGGER.info('clicking on Overview confirmed')
 
+    def test030_allow_unit_field_to_be_displayed_in_case_of_mibi(self):
+        """
+        New: Test unit: limit of quantification Approach: Allow the unit field to display when I select quantitative MiBi type & make sure it displayed in the active table & in the export sheet 
+        
+        Make sure the unit displayed in the active table & in the export sheet 
+        In case I create test unit with type quantitative MiBi, Unit field opened beside the upper limit & the concentration. 
+        """
+
+        testunit_record = self.test_unit_page.search(value='Quantitative MiBi')[0]
+        row_data = self.base_selenium.get_row_cells_dict_related_to_header(row=testunit_record)
+        testunit_number = row_data['Test Unit No.']
+        if row_data['Unit'] == '-':
+            self.base_selenium.LOGGER.info('unit field has no value, update the record to make sure ')
+            self.test_unit_page.open_edit_page(row=testunit_record)
+            random_unit = self.test_unit_page.generate_random_text()
+            self.test_unit_page.set_spec_unit(value=random_unit)
+            self.test_unit_page.save(save_btn='general:save_form', logger_msg='Save testunit')
+            self.test_unit_page.get_test_units_page()
+        
+        testunit_record = self.test_unit_page.search(value=testunit_number)[0]
+        row_data = self.base_selenium.get_row_cells_dict_related_to_header(row=testunit_record)
+
+        self.base_selenium.LOGGER.info('unit field has value {}'.format(row_data['Unit']))
+        if row_data['Unit'] != '-':
+            self.assertNotEqual(row_data['Unit'], '-')
+        else:
+            self.assertEqual(row_data['Unit'], random_unit)
```

Comment:
replace with
```
self.assertEqual(row_data['Unit'], random_unit)
```

Reply:
it adds random value only in case it doesn't already have value in the unit field, but case it already have value, i only need to make sure that it has unit column in the table and in the export